cdap-ui/app/cdap/components/Experiments/DetailedView/ModelsTable/index.js

```diff
@@ -118,153 +100,166 @@ const getNewHeadersBasedOnOutcome = (outcomeType) => (
+      {renderMetrics(newHeaders, model, experimentId)}
       {
-        newHeaders.map((tableHeader, i) => {
-          return (
-            <div
-              className="grid-header-item"
-              title={tableHeader.label}
-              key={i}
-              style={{ width: `${tableHeader.width}` }}
-            >
-              {
-                tableHeader.property ?
-                  renderSortableTableHeader(tableHeader)
-                :
-                  tableHeader.label
-              }
-            </div>
-          );
-        })
+        wrapContentWithTitleAttr(
+          <DeleteModelBtn
+            experimentId={experimentId}
+            model={model}
+          />
+        )
       }
-    </div>
+    </Component>
   );
 };
 
-const renderTableBody = (experimentId, outcomeType, models) => {
-  let list = models.map(model => {
+function renderGridBody(models, outcomeType, experimentId) {
+  let list = addDetailedModelObject([...models]);
+  return list.map((model) => {
     let {name, algorithm, hyperparameters} = model;
-    return {
+    model = {
       ...model,
       name,
       algorithmLabel: getAlgorithmLabel(algorithm),
       hyperparameters
     };
-  });
-  const renderItem = (width, content) => (
-    <div
-      className="grid-body-item"
-      title={isNumber(content) || isString(content) ? content : ''}
-      style={{ width: `${width}` }}
-    >
-      {content}
-    </div>
-  );
-  const renderMetrics = (newHeaders, model) => {
-    let metrics;
-    let len = newHeaders.length - 1;
-    let commonHeadersLen = tableHeaders.length - 1;
-    if (NUMBER_TYPES.indexOf(outcomeType) !== -1) {
-      metrics = newHeaders.slice(commonHeadersLen, len);
-    } else {
-      metrics = newHeaders.slice(commonHeadersLen, len);
+    if (model.detailedView) {
+      return renderModelDetails(model);
     }
-    return metrics.map(t => renderItem(t.width, model.evaluationMetrics[t.property] || '--'));
-  };
+    return renderModel(model, outcomeType, experimentId);
+ });
+}
 
+function renderGrid(models, outcomeType, experimentId) {
   let newHeaders = getNewHeadersBasedOnOutcome(outcomeType);
   return (
-    <div className="grid-body">
-      {
-        list.map((model) => {
-          let Component = 'div';
-          let props = {
-            className: classnames("grid-body-row-container", {
-              "opened": model.active
-            }),
-            key: model.id
-          };
-          let inSplitStep = (['SPLITTING', 'DATA_READY', 'EMPTY'].indexOf(model.status) !== -1);
-          if (inSplitStep) {
-            Component = Link;
-            props.to = `/ns/${getCurrentNamespace()}/experiments/create?experimentId=${experimentId}&modelId=${model.id}`;
+    <div className={classnames("grid grid-container", {
+      'classification': NUMBER_TYPES.indexOf(outcomeType) === -1
+    })}>
+      <div className="grid-header">
+        <div className="grid-item">
+          <strong></strong>
```

Comment:
Do the empty <strong> tags here and on line 254 serve any purpose?

Reply:
Yes. They take the empty column for the caret signs and the delete icons at the ends of each models row. If you see the `grid-template-columns` for each `grid-item`, we have it as `20px 1fr 1fr 2fr 1fr 1fr 1fr 1fr 20px;` The first 20px and the last 20px are for these.